### PR TITLE
NIFI-13751 Update com.hierynomus:sshj to v0.39.0

### PR DESCRIPTION
**Pull request by EndzeitBegins**
# Summary

[NIFI-13751](https://issues.apache.org/jira/browse/NIFI-13751)

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [x] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [x] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [x] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [x] Pull Request based on current revision of the `main` branch
- [x] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [x] Build completed using `mvn clean install -P contrib-check`
  - [x] JDK 21

### Licensing

- [x] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [x] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [x] Documentation formatting appears as expected in rendered files

### CODE REVIEW

File: nifi-extension-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/java/org/apache/nifi/processors/standard/util/TestSFTPTransfer.java [hunk 1]
```diff
@@ -262,7 +263,7 @@ public void testFileTypeUnknown() throws IOException {
         when(processContext.getProperty(eq(FileTransfer.REMOTE_PATH))).thenReturn(new MockPropertyValue("."));
 
         try (SFTPClient sftpClient = mock(SFTPClient.class)) {
-            when(sftpClient.ls(any(), any())).then(invocation -> {
+            when(sftpClient.ls(any(), ArgumentMatchers.<RemoteResourceFilter>any())).then(invocation -> {
```
Comment (exceptionfactory) [inline nifi-extension-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/java/org/apache/nifi/processors/standard/util/TestSFTPTransfer.java:266]: @EndzeitBegins is this addition required?

**Review by exceptionfactory — COMMENTED** [review]
Thanks for the upgrade @EndzeitBegins, just one question on the test method change.

File: nifi-extension-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/java/org/apache/nifi/processors/standard/util/TestSFTPTransfer.java (hunk 1 above)
Comment (EndzeitBegins) [reply nifi-extension-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/java/org/apache/nifi/processors/standard/util/TestSFTPTransfer.java:266]: Hey David. I'm afraid this change is required as [I introduced a new overload to `sshj`](https://github.com/hierynomus/sshj/pull/928) for `ls` in order to be able to `break` from listings on large directories. 
Up to `0.38.0` there were:
- `ls(String path)`, and 
- `ls(String path, RemoteResourceFilter filter)`.
With `0.39.0` the new overload `ls(String path, RemoteResourceSelector selector)` is added.

Due to this the compiler cannot infer the type for the second `any()` anymore.

**Review by exceptionfactory — APPROVED** [review]
Thanks again for the contribution, and particularly for the improvement upstream in SSHJ @EndzeitBegins! +1 merging
